src/docfx/lib/git/GitUtility.cs

```diff
@@ -44,286 +42,114 @@ public static string FindRepo(string path)
         }
 
         /// <summary>
-        /// Get git commits group by files
+        /// Clone git repository from remote to local
         /// </summary>
-        /// <param name="repoPath">The git repo root path</param>
-        /// <param name="files">The collection of git repo files</param>
-        /// <param name="progress">The processing progress</param>
-        /// <returns>A collection of git commits</returns>
-        public static unsafe List<GitCommit>[] GetCommits(string repoPath, List<string> files, Action<int, int> progress = null)
+        /// <param name="cwd">The current working directory</param>
+        /// <param name="remote">The remote url</param>
+        /// <param name="path">The path to clone</param>
+        /// <returns>Task status</returns>
+        public static Task Clone(string cwd, string remote, string path)
         {
-            Debug.Assert(files.All(file => !PathUtility.FilePathHasInvalidChars(file)));
-
-            var pathToParent = BuildPathToParentPath(files);
-            var pathToParentByRef = pathToParent.ToDictionary(p => p.Key, p => p.Value, RefComparer.Instance);
-            var repo = OpenRepo(repoPath);
-            var commits = LoadCommits(repo);
-            var trees = LoadTrees(repo, commits, pathToParent, progress);
+            Debug.Assert(PathUtility.FolderPathHasInvalidChars(path));
```

Comment:
>(P [](start = 24, length = 2)

!

Reply:
thanks!

---
In reply to: [180632701](https://github.com/dotnet/docfx/pull/2611#discussion_r180632701) [](ancestors = 180632701)

Reply:
others are checked in Execute Method

---
In reply to: [180636653](https://github.com/dotnet/docfx/pull/2611#discussion_r180636653) [](ancestors = 180636653,180632701)